web/src/components/organisms/Workspace/Settings/hooks.ts

```diff
@@ -0,0 +1,61 @@
+import { useCallback } from "react";
+import { useNavigate } from "react-router-dom";
+
+import Notification from "@reearth-cms/components/atoms/Notification";
+import {
+  useUpdateWorkspaceMutation,
+  useDeleteWorkspaceMutation,
+} from "@reearth-cms/gql/graphql-client-api";
+import { useT } from "@reearth-cms/i18n";
+import { useWorkspace } from "@reearth-cms/state";
+
+export default () => {
+  const navigate = useNavigate();
+  const [currentWorkspace, setCurrentWorkspace] = useWorkspace();
+  const t = useT();
+
+  const workspaceId = currentWorkspace?.id;
+  const workspaceName = currentWorkspace?.name;
+
+  const [updateWorkspaceMutation] = useUpdateWorkspaceMutation();
+  const [deleteWorkspaceMutation] = useDeleteWorkspaceMutation({
+    refetchQueries: ["GetMe"],
+  });
+
+  const handleWorkspaceUpdate = useCallback(
+    async (name?: string) => {
+      if (!workspaceId || !name) return;
+      const res = await updateWorkspaceMutation({
+        variables: {
+          workspaceId,
+          name,
+        },
+      });
+      if (res.errors || !res.data?.updateWorkspace) {
+        Notification.error({ message: t("Failed to update workspace.") });
+      } else {
+        setCurrentWorkspace({ ...res.data.updateWorkspace.workspace });
```

Comment:
Can't you just do `setCurrentWorkspace(res.data.update.workspace.workspace)`?

Reply:
Still not addressed

Reply:
@KaWaite , yes it can be done, updated.